src/main/java/cn/nukkit/block/BlockBed.java

```diff
@@ -43,6 +48,10 @@
     @Since("1.4.0.0-PN")
     public static final BlockProperties PROPERTIES = new BlockProperties(DIRECTION, OCCUPIED, HEAD_PIECE);
 
+    public static final List<Integer> preventSleepMonsterId = Arrays.asList(EntityBlaze.NETWORK_ID, EntityCaveSpider.NETWORK_ID, EntityCreeper.NETWORK_ID, EntityDrowned.NETWORK_ID, EntityElderGuardian.NETWORK_ID, EntityEndermite.NETWORK_ID, EntityEvoker.NETWORK_ID, EntityGuardian.NETWORK_ID, EntityHusk.NETWORK_ID, EntityPiglinBrute.NETWORK_ID, EntityPillager.NETWORK_ID, EntityPhantom.NETWORK_ID, EntityRavager.NETWORK_ID, EntitySilverfish.NETWORK_ID, EntitySkeleton.NETWORK_ID, EntityStray.NETWORK_ID, EntityVex.NETWORK_ID, EntityVindicator.NETWORK_ID, EntityWitch.NETWORK_ID, EntityWither.NETWORK_ID, EntityWitherSkeleton.NETWORK_ID, EntityZoglin.NETWORK_ID, EntityZombie.NETWORK_ID, EntityZombieVillager.NETWORK_ID); // TODO: Check Spider Jockey
+
+    public static final List<Integer> preventSleepOnlyHostileMonsterId = Arrays.asList(EntityEnderman.NETWORK_ID, EntityZombiePigman.NETWORK_ID); // TODO: Check Chicken Jockey
+
```

Comment:
I think it would be better to add a method like `Entity.isPreventingSleep(Player)` returning `false` by default and override it to `true` on the entities that should prevent sleep, in case of the Enderman it could return `true` only when it is aggroed

Reply:
thank you